extensions/functions_datetime.yaml

```diff
@@ -6,25 +6,49 @@ scalar_functions:
     description: Extract portion of a date/time value.
     impls:
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ YEAR, ISO_YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, ISO_DAY_OF_WEEK, DAY_OF_YEAR, WEEK,
+                       ISO_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND, SUBSECOND, EPOCH ]
             name: The part of the value to extract.
             required: true
           - value: timestamp
         return: i64
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ YEAR, ISO_YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, ISO_DAY_OF_WEEK, DAY_OF_YEAR, WEEK,
+                       ISO_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND, SUBSECOND, EPOCH,
+                       TIMEZONE_OFFSET ]
```

Comment:
- It's not clear to me what `MILLISECOND`, `MICROSECOND` and `NANOSECOND` are relative to. Does for example `extract(MICROSECOND, 2022-08-15T11:08:24.123456789)` yield 24123456, 123456, or 456?
 - What does `SUBSECOND` mean?
 - What does `EPOCH` mean?
 - I'm not 100% convinced `TIMEZONE_OFFSET` belongs here, even if we keep the timezone string argument. Is that supposed to depend on the timestamp due to regulatory changes, DST, and whatnot? If so, can we *please* restrict that nonsense to dedicated functions converting between `timestamp` and `timestamp_tz` instead? If not, the result would not depend on the `timestamp_tz` value.
 - If we keep it, with what unit are `TIMEZONE_OFFSET`s represented (minutes? seconds?)
 - What's the difference between `YEAR` and `ISO_YEAR`?
 - Do `MONTH`, `DAY`, ..., `ISO_WEEK` start counting at 1 or 0?
 - Which day is the first day of the week?

Please elaborate in the function description; it should probably have an exhaustive description of all the options, including the ones that were already defined prior.

Reply:
I've added a description to `extract` function that should address all questions above.
`TIMEZONE_OFFSET` should return seconds because that's the precision provided by the [tzdb](https://github.com/eggert/tz).

Reply:
Well perhaps microseconds could work for `TIMEZONE_OFFSET` too since everything is microseconds. I don't have a preference, leaving seconds for now.